### PR TITLE
TIP-1193 Use debug env vars instead of parameters

### PR DESCRIPTION
**Pull request by arnolanglade**
**Description (for Contributor and Core Developer)**

We've done a [preview PR](https://github.com/akeneo/pim-community-dev/pull/10460) about loading env vars following the new symfony standard. The purpose of this PR is to use them in the PIM now.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: .env.dist [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+APP_DATABASE_NAME=akeneo_pim
+APP_DATABASE_USER=akeneo_pim
+APP_DATABASE_PASSWORD=akeneo_pim
+APP_DEFAULT_LOCALE=en
+APP_SECRET=ThisTokenIsNotSoSecretChangeIt
+APP_INDEX_HOSTS='elastic:changeme@elasticsearch:9200'
+APP_PRODUCT_INDEX_NAME=akeneo_pim_product
+APP_PRODUCT_PROPOSAL_INDEX_NAME=akeneo_pim_product_proposal
+APP_PUBLISHED_PRODUCT_INDEX_NAME=akeneo_pim_published_product
+APP_PUBLISHED_PRODUCT_AND_PRODUCT_MODEL_INDEX_NAME=akeneo_pim_published_product_and_product_model
```
Comment (ahocquard) [inline .env.dist]: same: only product_and_product_model_index exist now. Also, it's useless in CE.

File: .env.dist [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
+APP_PRODUCT_MODEL_INDEX_NAME=akeneo_pim_product_model
```
Comment (ahocquard) [inline .env.dist]: same

File: .env.dist [hunk 3]
```diff
@@ -1,6 +1,20 @@
...
+APP_ENV=prod
+APP_DATABASE_HOST=mysql
+APP_DATABASE_PORT=null
+APP_DATABASE_NAME=akeneo_pim
+APP_DATABASE_USER=akeneo_pim
+APP_DATABASE_PASSWORD=akeneo_pim
+APP_DEFAULT_LOCALE=en
+APP_SECRET=ThisTokenIsNotSoSecretChangeIt
+APP_INDEX_HOSTS='elastic:changeme@elasticsearch:9200'
+APP_PRODUCT_INDEX_NAME=akeneo_pim_product
```
Comment (ahocquard) [inline .env.dist]: I think this is useless. It should not exist anymore as we merged the index. We still have dead code

File: .env.dist [hunk 4]
```diff
@@ -1,6 +1,20 @@
...
+APP_PRODUCT_PROPOSAL_INDEX_NAME=akeneo_pim_product_proposal
+APP_PUBLISHED_PRODUCT_INDEX_NAME=akeneo_pim_published_product
+APP_PUBLISHED_PRODUCT_AND_PRODUCT_MODEL_INDEX_NAME=akeneo_pim_published_product_and_product_model
+APP_PRODUCT_MODEL_INDEX_NAME=akeneo_pim_product_model
+APP_PRODUCT_AND_PRODUCT_MODEL_INDEX_NAME=akeneo_pim_product_and_product_model
+APP_RECORD_INDEX_NAME=akeneo_referenceentity_record
```
Comment (ahocquard) [inline .env.dist]: useless in CE

File: .env.dist [hunk 5]
```diff
@@ -1,6 +1,20 @@
...
+APP_ASSET_INDEX_NAME=akeneo_assetmanager_asset
```
Comment (ahocquard) [inline .env.dist]: useless in CE

File: app/config/config.yml [hunk 6]
```diff
@@ -60,3 +53,12 @@ akeneo_pim_structure:
 
 parameters:
     pim_ce_dev_src_folder_location: '%kernel.project_dir%'
+    index_hosts: '%env(APP_INDEX_HOSTS)%'
+    product_index_name: '%env(APP_PRODUCT_INDEX_NAME)%'
+    product_proposal_index_name: '%env(APP_PRODUCT_PROPOSAL_INDEX_NAME)%'
```
Comment (ahocquard) [inline app/config/config.yml]: useless: product_proposal_index_name , published_product_index_name, published_product_and_product_model_index_name, product_model_index_name, record_index_name, asset_index_name in CE

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/config/services.yml [hunk 7]
```diff
@@ -14,7 +14,7 @@ services:
         class: 'Akeneo\Platform\Bundle\UIBundle\Resolver\LocaleResolver'
```
Comment (ahocquard) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/config/services.yml]: does it handle the case where "APP_DEFAULT_LOCALE" is not defined?

**Comment by jjanvier** [comment]
What you mean @ahocquard? There are default values in the `env` files.

**Comment by ahocquard** [comment]
@jjanvier when you create s skeleton of a symfony app, it is recommended to do it this way:

```
# Put parameters here that don't need to change on each machine where the app is deployed
# https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration
parameters:
    env(API_URL): 'Your forgot to set API_URL environment variable.'
    env(API_USER): 'Your forgot to set API_USER environment variable.'
    env(API_PASSWORD): 'Your forgot to set API_PASSWORD environment variable.'
    env(API_CLIENT): 'Your forgot to set API_CLIENT environment variable.'
```

At least, you have a value for the environment variable. Otherwise, you could have exception in some libraries that does not handle properly a missing environment variable.

https://symfony.com/doc/current/configuration.html#configuration-based-on-environment-variables

```
The next step is to define the value of those env vars in your shell, your web server, etc. This is explained in the following sections, but to protect your application from undefined env vars, you can give them a default value using the parameters key:
``` 

Personally, I did it in Symfony 4 projects I did.

**Comment by arnolanglade** [comment]
This is weird because the default values are defined in `.env.dist`

**Comment by arnolanglade** [comment]
https://twitter.com/arnolanglade/status/1156107440775192576

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/config/services.yml (hunk 7 above)
Comment (arnolanglade) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/config/services.yml]: you should have look to my last comment :)

**Comment by ahocquard** [comment]
Actually, it can lead to errors in production environment, as ".env" does not exist.

**Comment by arnolanglade** [comment]
No because it loads the `.env.dist` if `.env` does not exist. You should have a look to [bootstrap.php](https://github.com/akeneo/pim-enterprise-dev/blob/master/app/bootstrap.php)

**Comment by ahocquard** [comment]
@arnolanglade :+1:  I didn't know it thanks.

**Pull request merged**